### ISSUE TITLE
Segmentation mask includes incorrect pixels at bounding box edge (mask spill over the bounding box border)

### ISSUE THREAD

**Issue by GuptaKishan** [ultralytics/ultralytics#21088]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

After fine-tuning yolo11x-seg.pt on my custom dataset (pallet detection), I noticed that the predicted segmentation masks always leak or include incorrect pixels at the edge where the bounding box touches the mask boundary. This results in noisy pixels outside the true object shape.

Expected behavior:
The mask should follow the true contour of the object and not spill out where the bounding box touches the mask. This is critical in high-precision applications like volume estimation.

### Environment

Ultralytics 8.3.153 🚀 Python-3.10.12 torch-2.5.1+cu124 CUDA:0 (NVIDIA RTX A2000 12GB, 11907MiB)
Setup complete ✅ (32 CPUs, 62.5 GB RAM, 289.4/431.9 GB disk)

OS                  Linux-5.15.0-135-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             git
Path                /home/doks/repos/volume_estimation/venv/lib/python3.10/site-packages/ultralytics
RAM                 62.51 GB
Disk                289.4/431.9 GB
CPU                 13th Gen Intel Core(TM) i9-13900E
CPU count           32
GPU                 NVIDIA RTX A2000 12GB, 11907MiB
GPU count           1
CUDA                12.4

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.0.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.0>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
ultralytics-thop    ✅ 2.0.14>=2.0.0

### Minimal Reproducible Example

yolo segment train \
  model=yolo11x-seg.pt \
  data=data.yaml \
  epochs=200 \
  batch=16 \
  imgsz=1024 \
  optimizer=SGD \
  patience=20 \
  val=True

model = YOLO("fine_tuned_model_weights_path") 
results = model(image_path)
mask_image_with_bb = results[0].plot(boxes = True)
mask_image_without_bb = results[0].plot(boxes = False)

masks = results[0].masks.data.cpu().numpy() 
combined_mask = np.any(masks, axis=0).astype(np.uint8)
image_size = (1224, 1024)
resized_mask = cv2.resize(combined_mask, image_size, interpolation=cv2.INTER_NEAREST)
binary_mask_image_without_bb = resized_mask * 255

### Additional

![Image](https://github.com/user-attachments/assets/f6e1af92-a2cb-4f34-95d4-e1c4beadb359)

![Image](https://github.com/user-attachments/assets/90429a60-d201-47fb-8988-b1b4083874d3)

![Image](https://github.com/user-attachments/assets/64291a68-de39-419f-87e4-2e55fe12f634)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GuptaKishan, thank you for submitting your detailed report and for using Ultralytics 🚀! This is an automated response to help streamline your issue, and an Ultralytics engineer will review and assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for helpful guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, as well as our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Since this is a 🐛 Bug Report:  
Thank you for providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) and clear visualizations—this is very helpful for debugging!

Join the Ultralytics community to connect and share insights:
- For real-time chat, join our [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth questions, visit our [Discourse](https://community.ultralytics.com/)
- For knowledge sharing and discussions, check out our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you are using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), as this can resolve many issues:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments, with all dependencies (including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. The CI system verifies correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your report and patience while we investigate!

**Comment by Y-T-G** [comment]
You can use `retina_masks=True` in `model.predict()`

**Comment by GuptaKishan** [comment]
Thank you. It worked.

**Comment by glenn-jocher** [comment]
Great to hear that `retina_masks=True` resolved the mask spillover issue! This parameter ensures that segmentation masks are generated at full image resolution rather than being constrained by the bounding box dimensions, which eliminates the edge artifacts you were experiencing. This is particularly important for precision applications like volume estimation where accurate mask boundaries are critical.
